src/mjolnir/landmark_database_builder.cc

```diff
@@ -0,0 +1,254 @@
+  } else {
+    sqlite3_bind_null(insert_stmt, 1);
+  }
+
+  if (landmark.type != "") {
+    sqlite3_bind_text(insert_stmt, 2, landmark.type.c_str(), landmark.type.length(), SQLITE_STATIC);
+  } else {
+    sqlite3_bind_null(insert_stmt, 2);
+  }
+
+  sqlite3_bind_double(insert_stmt, 3, landmark.lng);
+  sqlite3_bind_double(insert_stmt, 4, landmark.lat);
+
+  LOG_INFO(sqlite3_expanded_sql(insert_stmt));
+
+  ret = sqlite3_step(insert_stmt);
+  if (ret == SQLITE_DONE || ret == SQLITE_ROW) {
+    LOG_INFO("inserted landmark");
+    did_inserts = true;
+    return true;
+  }
+
+  LOG_ERROR("sqlite3_step() error: " + std::string(sqlite3_errmsg(db)));
+  LOG_ERROR("sqlite3_step() Name: " + landmark.name);
+  LOG_ERROR("sqlite3_step() Type: " + landmark.type);
+  LOG_ERROR("sqlite3_step() longitude: " + std::to_string(landmark.lng));
+  LOG_ERROR("sqlite3_step() Latitude: " + std::to_string(landmark.lat));
+
+  return false;
+}
+
+bool LandmarkDatabase::prepare_bounding_box_stmt() {
+  sql =
+      "SELECT name, type, X(geom), Y(geom) FROM landmarks WHERE ST_Covers(BuildMbr(?, ?, ?, ?, 4326), geom)";
+
+  ret = sqlite3_prepare_v2(db, sql.c_str(), strlen(sql.c_str()), &bounding_box_stmt, NULL);
+  if (ret != SQLITE_OK) {
+    LOG_ERROR("SQL error: " + sql);
+    LOG_ERROR(std::string(sqlite3_errmsg(db)));
+    sqlite3_close(db);
+    return false;
+  }
+
+  LOG_INFO("prepared bounding box statement");
+  return true;
+}
+
+bool LandmarkDatabase::get_landmarks_in_bounding_box(std::vector<Landmark>* landmarks,
+                                                     const double minLat,
+                                                     const double minLong,
+                                                     const double maxLat,
+                                                     const double maxLong) {
+  sqlite3_reset(bounding_box_stmt);
+  sqlite3_clear_bindings(bounding_box_stmt);
+
+  sqlite3_bind_double(bounding_box_stmt, 1, minLong);
+  sqlite3_bind_double(bounding_box_stmt, 2, minLat);
+  sqlite3_bind_double(bounding_box_stmt, 3, maxLong);
+  sqlite3_bind_double(bounding_box_stmt, 4, maxLat);
+
+  LOG_INFO(sqlite3_expanded_sql(bounding_box_stmt));
+
+  ret = sqlite3_step(bounding_box_stmt);
+  while (ret == SQLITE_ROW) {
+    const char* name = reinterpret_cast<const char*>(sqlite3_column_text(bounding_box_stmt, 0));
+    const char* type = reinterpret_cast<const char*>(sqlite3_column_text(bounding_box_stmt, 1));
+    double lng = sqlite3_column_double(bounding_box_stmt, 2);
+    double lat = sqlite3_column_double(bounding_box_stmt, 3);
+    landmarks->emplace_back(Landmark{name, type, lng, lat});
+
+    ret = sqlite3_step(bounding_box_stmt);
+  }
+
+  if (ret == SQLITE_DONE || SQLITE_OK) {
+    return true;
+  }
+
+  LOG_ERROR("Error: " + std::string(sqlite3_errmsg(db)));
+  sqlite3_close(db);
+  return false;
+}
+
+void LandmarkDatabase::close_database() {
+  sqlite3_close(db);
```

Comment:
i think you need to release your prepared statments here too, i forget the function call but i remember it exists

Reply:
`sqlite3_finalize`